tests/bsim/bluetooth/mesh/src/test_sar.c

```diff
@@ -136,22 +136,22 @@ static const struct bt_mesh_model_op _dummy_vnd_mod_op[] = {
 
 uint16_t dummy_keys[CONFIG_BT_MESH_MODEL_KEY_COUNT] = { 0 };
 
-static struct bt_mesh_model dummy_vnd_mod = {
+static const struct bt_mesh_model dummy_vnd_mod = {
 	.op = _dummy_vnd_mod_op,
 	.keys = dummy_keys,
 	.keys_cnt = CONFIG_BT_MESH_MODEL_KEY_COUNT,
 	.vnd.id = TEST_VND_MOD_ID,
 	.vnd.company = TEST_VND_COMPANY_ID,
 };
 
-static struct bt_mesh_elem elements[] = {BT_MESH_ELEM(
+static const struct bt_mesh_elem elements[] = {BT_MESH_ELEM(
 	0,
 	MODEL_LIST(BT_MESH_MODEL_CFG_SRV,
 		   BT_MESH_MODEL_CFG_CLI(&cfg_cli),
 		   BT_MESH_MODEL_SAR_CFG_CLI(&sar_cli),
 		   BT_MESH_MODEL_SAR_CFG_SRV),
 	MODEL_LIST(BT_MESH_MODEL_VND_CB(TEST_VND_COMPANY_ID, TEST_VND_MOD_ID, _dummy_vnd_mod_op,
-					NULL, &dummy_vnd_mod, NULL)))};
+					NULL, (void *)&dummy_vnd_mod, NULL)))};
```

Comment:
Why is it required to cast to void pointer?

Reply:
will be warning if not cast to void

Reply:
I clearly understand the reason is the warning. Probably anything went wrong with the initial types since the warning appeared here. Seems like a workaround. The initial reason for the warning should be fixed.